File: templates/problems/04_stack/03_eval_rpn/eval_rpn_test.go
package eval_rpn

import "testing"

func TestEvalRPN(t *testing.T) {
	tests := []struct {
		name   string
		tokens []string
		want   int
	}{
		{"simple addition", []string{"2", "1", "+"}, 3},
		{"simple subtraction", []string{"4", "2", "-"}, 2},
		{"simple multiplication", []string{"3", "4", "*"}, 12},
		{"simple division", []string{"6", "3", "/"}, 2},
		{"complex expression", []string{"2", "1", "+", "3", "*"}, 9},
		{"leetcode example", []string{"4", "13", "5", "/", "+"}, 6},
		{"negative result", []string{"1", "2", "-"}, -1},
		{"single number", []string{"42"}, 42},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			if got := evalRPN(tt.tokens); got != tt.want {
				t.Errorf("evalRPN(%v) = %d, want %d", tt.tokens, got, tt.want)
			}
		})
	}
}

File: templates/problems/04_stack/03_eval_rpn/typescript/evalRpn.test.ts
import { describe, it, expect } from 'vitest';
import { evalRPN } from './evalRpn';

describe('EvalRpn', () => {
  it('simple addition', () => {
    expect(evalRPN(['2', '1', '+'])).toBe(3);
  });

  it('simple subtraction', () => {
    expect(evalRPN(['4', '2', '-'])).toBe(2);
  });

  it('simple multiplication', () => {
    expect(evalRPN(['3', '4', '*'])).toBe(12);
  });

  it('simple division', () => {
    expect(evalRPN(['6', '3', '/'])).toBe(2);
  });

  it('complex expression', () => {
    expect(evalRPN(['2', '1', '+', '3', '*'])).toBe(9);
  });

  it('leetcode example', () => {
    expect(evalRPN(['4', '13', '5', '/', '+'])).toBe(6);
  });

  it('negative result', () => {
    expect(evalRPN(['1', '2', '-'])).toBe(-1);
  });

  it('single number', () => {
    expect(evalRPN(['42'])).toBe(42);
  });
});
